src/write_outputs/write_subsidy_revenue.jl

```diff
@@ -9,13 +9,55 @@ function write_subsidy_revenue(path::AbstractString, inputs::Dict, setup::Dict,
 
 	dfSubRevenue = DataFrame(Region = dfGen[!, :region], Resource = inputs["RESOURCES"], Zone = dfGen[!, :Zone], Cluster = dfGen[!, :cluster], R_ID=dfGen[!, :R_ID], SubsidyRevenue = zeros(G))
 	MIN_CAP = dfGen[(dfGen[!, :Min_Cap_MW].>0), :R_ID]
+	if !isempty(inputs["VRE_STOR"])
+		dfVRE_STOR = inputs["dfVRE_STOR"]
+		MIN_CAP_SOLAR = dfVRE_STOR[(dfVRE_STOR[!, :Min_Cap_Solar_MW].>0), :R_ID]
+		MIN_CAP_WIND = dfVRE_STOR[(dfVRE_STOR[!, :Min_Cap_Wind_MW].>0), :R_ID]
+		MIN_CAP_STOR = dfGen[(dfGen[!, :Min_Cap_MWh].>0), :R_ID]
+		if !isempty(MIN_CAP_SOLAR)
+			dfSubRevenue.SubsidyRevenue[MIN_CAP_SOLAR] .+= (value.(EP[:eTotalCap_SOLAR])[MIN_CAP_SOLAR]) .* (dual.(EP[:cMinCap_Solar][MIN_CAP_SOLAR])).data
+		end
+		if !isempty(MIN_CAP_WIND)
+			dfSubRevenue.SubsidyRevenue[MIN_CAP_WIND] .+= (value.(EP[:eTotalCap_WIND])[MIN_CAP_WIND]) .* (dual.(EP[:cMinCap_Wind][MIN_CAP_WIND])).data
+		end
+		if !isempty(MIN_CAP_STOR)
+			dfSubRevenue.SubsidyRevenue[MIN_CAP_STOR] .+= (value.(EP[:eTotalCap_STOR])[MIN_CAP_STOR]) .* (dual.(EP[:cMinCap_Stor][MIN_CAP_STOR])).data
+		end
+	end
 	dfSubRevenue.SubsidyRevenue[MIN_CAP] .= (value.(EP[:eTotalCap])[MIN_CAP]) .* (dual.(EP[:cMinCap][MIN_CAP])).data
 	### calculating tech specific subsidy revenue
 	dfRegSubRevenue = DataFrame(Region = dfGen[!, :region], Resource = inputs["RESOURCES"], Zone = dfGen[!, :Zone], Cluster = dfGen[!, :cluster], R_ID=dfGen[!, :R_ID], SubsidyRevenue = zeros(G))
 	if (setup["MinCapReq"] >= 1)
 		for mincap in 1:inputs["NumberOfMinCapReqs"] # This key only exists if MinCapReq >= 1, so we can't get it at the top outside of this condition.
 			MIN_CAP_GEN = dfGen[(dfGen[!, Symbol("MinCapTag_$mincap")].==1), :R_ID]
 			dfRegSubRevenue.SubsidyRevenue[MIN_CAP_GEN] .= dfRegSubRevenue.SubsidyRevenue[MIN_CAP_GEN] + (value.(EP[:eTotalCap][MIN_CAP_GEN])) * (dual.(EP[:cZoneMinCapReq][mincap]))
+			if !isempty(inputs["VRE_STOR"])
+				MIN_CAP_GEN_SOLAR = dfVRE_STOR[(dfVRE_STOR[!, Symbol("MinCapTagSolar_$mincap")].==1), :R_ID]
+				MIN_CAP_GEN_WIND = dfVRE_STOR[(dfVRE_STOR[!, Symbol("MinCapTagWind_$mincap")].==1), :R_ID]
+				MIN_CAP_GEN_ASYM_DC_DIS = intersect(inputs["VS_ASYM_DC_DISCHARGE"], dfVRE_STOR[(dfVRE_STOR[!, Symbol("MinCapTagStor_$mincap")].==1), :R_ID])
+				MIN_CAP_GEN_ASYM_AC_DIS = intersect(inputs["VS_ASYM_AC_DISCHARGE"], dfVRE_STOR[(dfVRE_STOR[!, Symbol("MinCapTagStor_$mincap")].==1), :R_ID])
+				MIN_CAP_GEN_SYM_DC = intersect(inputs["VS_SYM_DC"], dfVRE_STOR[(dfVRE_STOR[!, Symbol("MinCapTagStor_$mincap")].==1), :R_ID])
+				MIN_CAP_GEN_SYM_AC = intersect(inputs["VS_SYM_AC"], dfVRE_STOR[(dfVRE_STOR[!, Symbol("MinCapTagStor_$mincap")].==1), :R_ID])
+				if !isempty(MIN_CAP_GEN_SOLAR)
+					dfRegSubRevenue.SubsidyRevenue[MIN_CAP_GEN_SOLAR] .= dfRegSubRevenue.SubsidyRevenue[MIN_CAP_GEN_SOLAR] + (value.(EP[:eTotalCap_SOLAR][MIN_CAP_GEN_SOLAR]).data) .* dfVRE_STOR[((dfVRE_STOR[!, Symbol("MinCapTagSolar_$mincap")].==1)), :EtaInverter] * (dual.(EP[:cZoneMinCapReq][mincap]))
```

Comment:
These lines are pretty long. We could easily shorten them by splitting out the Symbol-making to lines near the start of the for-loop or just inside `!isempty(inputs["VRE_STOR"])`.